### PR TITLE
Add web nfc support

### PR DESCRIPTION
**Pull request by odinuge**
This adds web nfc support on modern Android devices. Registers the same
keys as the web serial one, with a few exceptions. Most should be
possible to deal with tho.

Our white cards works similar on both, and the data format is the same.

Makes it way easier to host a vote, since you only need one android device with nfc. Android devices can also be connected to an external monitor as well, making them a cheap and easy solution :100: 

Example of web nfc: https://googlechrome.github.io/samples/web-nfc/ 

### TODO
- [ ] ~Separate impl~ (Not sure we should/need to do it)
- [x] Make better gui for selecting input. (Nah, this works I guess)
- [x] Fix handling of NTNU-style cards in the old impl. Works in this one.
- [ ] Error handling when "bad" cards are being used (out of scope of this PR)
- [x] Make web serial and web nfc work with the same set of cards.

### CODE REVIEW

**Comment by odinuge** [comment]
Hmm. Maybe we can do the same as online/dotkom, and use this for NTNU cards for abacard? :thinking:

**Comment by LudvigHz** [comment]
> Hmm. Maybe we can do the same as online/dotkom, and use this for NTNU cards for abacard? thinking

Ooh, not a bad idea :open_mouth:

**Comment by SmithPeder** [comment]
>Hmm. Maybe we can do the same as online/dotkom, and use this for NTNU cards for Abacard? 🤔

I like the idea of **not** having to use the white cards but one might have to explain to people that their cards are not linked to the vote. It's the exact same "_security_" measures as with the other cards, but people might think their card/user is linked to their vote in some way.

In addition, it should also work for the white cards in case people forget their NTNU card 🙂

**Comment by odinuge** [comment]
> I like the idea of not having to use the white cards but one might have to explain to people that their cards are not linked to the vote. It's the exact same "security" measures as with the other cards, but people might think their card/user is linked to their vote in some way.

I was talking about Abacard on Abakus.no, not VOTE :sweat_smile: 

Both the white cards and NTNU cards works now for both the old readers and nfc (with this and #380).

**Comment by SmithPeder** [comment]
>I was talking about Abacard on Abakus.no, not VOTE 😅

Oh lol, I just default to Vote since it was this repo 🙃  Didn't read the sentence close enough...

**Comment by odinuge** [comment]
Also, when we can use web nfc, we can have more than two scanner positions on genfors, making it faster :100:

**Comment by odinuge** [comment]
Screenshot_20210113-123243

**Comment by SmithPeder** [comment]
The `lint:prettier` wont use prettierignore the when there is a --ignore-path i think.
```json
//package.json
"lint:prettier": "prettier '**/*.{js,pug}' --list-different --ignore-path .gitignore",
```
I did the same mistake i `:develop`. https://github.com/webkom/vote/blob/develop/.prettierignore

**Comment by odinuge** [comment]
Ahh, true! Just got tierd of `yarn prettier` formatting all js bundles :sweat_smile:

**Comment by odinuge** [comment]
Removed the commit now

**Comment by odinuge** [comment]
Care to look at this before genfors @LudvigHz @SmithPeder? If we don't merge it now, it will become stall and i'll just close the PR i guess...

**Review by LudvigHz — APPROVED** [review]
LGTM:shipit:
Tested on my own device and works great.

File: client/services/cardKeyService.js [hunk 1]
```diff
@@ -74,64 +74,116 @@ module.exports = [
...
+          // Constantly send the readCardCommand and read the result.
+          // If there is no card, the result will be an error status,
+          // which is handled in the onComplete function
+          const runPoll = async () => {
+            try {
+              $rootScope.serialDevice.writer.write(readCardCommand);
+              await readResult();
+            } catch (e) {
+              /* eslint no-console: 0 */
+              console.err('Error doing card stuff', e);
```
Comment (SmithPeder) [inline client/services/cardKeyService.js:180]: ⛔ 
image

**Review by SmithPeder — CHANGES_REQUESTED** [review]

File: client/services/cardKeyService.js (hunk 1 above)
Comment (odinuge) [reply client/services/cardKeyService.js]: :shipit:

Comment (odinuge) [reply client/services/cardKeyService.js]: How did you get those errors?

**Review by SmithPeder — APPROVED** [review]

**Pull request merged**